### PR TITLE
feat(insights): add insights support to Hits widget

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This integrates insights with Angular Instantsearch to help using Click Analytics in Hits.

Using `connectHitsWithInsights` instead of `connectHits` exposed a new property `insights` to the internal state, which user can call to emit insights events.

**Result**

```html
<ais-hits>
    <ng-template let-hits="hits" let-insights="insights">
      <div *ngFor="let hit of hits">
        <ais-highlight attribute="name" [hit]="hit"></ais-highlight>

        <button (click)="insights('clickedObjectIDsAfterSearch', { eventName: 'Add to cart', objectIDs: [hit.objectID] })">
          Add to favorite
        </button>
      </div>
    </ng-template>
</ais-hits>
```

### CODE REVIEW

File: examples/storybook/src/wrap-with-hits.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@ type WrapWithHitsParams = {
   methods?: {};
   searchFunction?: (helper: Helper) => void;
   searchClient?: {};
+  insightsClient?: (method: string, payload: object) => void;
```
Comment (samouss) [inline examples/storybook/src/wrap-with-hits.ts]: We can add a `@TODO` comment to not forget to update type with the one from InstantSearch once available.

File: src/hits/__tests__/hits.spec.ts [hunk 2]
```diff
@@ -87,4 +87,40 @@ describe('Hits', () => {
...
+            </ng-template>
+          </ais-hits>
+        `,
+      TestedWidget: NgAisHits,
+      additionalDeclarations: [NgAisHighlight],
+    });
+
+    const insights = jest.fn();
+    const fixture = render({ insights });
+    expect(fixture).toMatchSnapshot();
```
Comment (samouss) [inline src/hits/__tests__/hits.spec.ts]: The snapshot is not really useful for this assertion, is it?

Comment (tkrugg) [reply src/hits/__tests__/hits.spec.ts]: 👍  I've removed it

File: examples/storybook/src/wrap-with-hits.ts (hunk 1 above)
Comment (tkrugg) [reply examples/storybook/src/wrap-with-hits.ts]: 👍 it's been added

**Pull request merged**